setup/mail-dovecot.sh

```diff
@@ -66,7 +66,33 @@ tools/editconf.py /etc/dovecot/conf.d/10-mail.conf \
 	first_valid_uid=0
 
 # Create, subscribe, and mark as special folders: INBOX, Drafts, Sent, Trash, Spam and Archive.
-cp conf/dovecot-mailboxes.conf /etc/dovecot/conf.d/15-mailboxes.conf
+cp conf/dovecot/conf.d/15-mailboxes.conf /etc/dovecot/conf.d/
+sed -i "s/#mail_plugins =\(.*\)/mail_plugins =\1 \$mail_plugins quota/" /etc/dovecot/conf.d/10-mail.conf
+if ! grep -q "mail_plugins.* imap_quota" /etc/dovecot/conf.d/20-imap.conf; then
+  sed -i "s/\(mail_plugins =.*\)/\1\n  mail_plugins = \$mail_plugins imap_quota/" /etc/dovecot/conf.d/20-imap.conf
+fi
+
+# configure stuff for quota support
+if ! grep -q "quota_status_success = DUNNO" /etc/dovecot/conf.d/90-quota.conf; then
+    cat > /etc/dovecot/conf.d/90-quota.conf << EOF;
+plugin {
+  quota = maildir
```

Comment:
There is a newer quota backend called "count" that is recommended. I'm reluctant to introduce a new feature that uses an out of date practice.

Reply:
Definitely gonna take a bit of investigation to make this happen.  

The count backend: 
 https://github.com/dovecot/core/blob/076cf225aa8d18e8abb0cbdc04f68abdf1c62a5e/src/plugins/quota/quota-count.c#L8
 
The maildir backend:
https://github.com/dovecot/core/blob/076cf225aa8d18e8abb0cbdc04f68abdf1c62a5e/src/plugins/quota/quota-maildir.c

My C is quite rusty (no pun intended), so getting this in will be a bit of a trick.  I have a path forward that would involve switching to the new backend, seeing what breaks, and then debugging it by looking at the C code and the database.

Reply:
See https://github.com/mail-in-a-box/mailinabox/pull/2387#issuecomment-2226135899 for why I think it may be fine to leave the maildir backend